### ISSUE TITLE
Sketchy Loss Functions

### ISSUE THREAD

**Issue by twhughes** [fancompute/neuroptica#4]
When I train simple linear models, the loss function oscillates wildly.  For example, using `inSituAdam`:

<img width="375" alt="image" src="https://user-images.githubusercontent.com/5669557/49624797-bd803f80-f987-11e8-803d-638292b8e76b.png">

```
model_linear = neu.Sequential([
    neu.ClementsLayer(N),
    neu.Activation(neu.Abs(N)),
    neu.DropMask(N, keep_ports=range(N_classes))
])

losses = neu.InSituAdam(model_linear, neu.CategoricalCrossEntropy, step_size=step_size).fit(x_train_flattened, y_train_onehot, epochs=n_epochs, batch_size=batch_size)
```

This may be a sign that the gradients are incorrect.  Should double check.
